src/accounts_refactor.rs

```diff
@@ -0,0 +1,339 @@
+    pub(crate) async fn update_scan(
+        mut self,
+        event: LedgerEvent<L>,
+        source: EventSource,
+        records_commitment: MerkleCommitment,
+    ) -> Result<AccountEditor<'a, L, Key>, KeystoreError<L>> {
+        let mut scan = match self.account.scan.take() {
+            Some(scan) => scan,
+            None => return Err(KeystoreError::ScanNotFound),
+        };
+        scan.handle_event(event, source);
+        // Check if the scan is complete.
+        if let Err(scan) = scan.finalize(records_commitment) {
+            self.account.scan = Some(scan);
+        }
+        Ok(self)
+    }
+
+    /// Save the account to the store.
+    ///
+    /// Returns the stored account.
+    pub fn save(&mut self) -> Result<Account<L, Key>, KeystoreError<L>> {
+        self.store.store(&self.account.pub_key(), &self.account)?;
+        Ok(self.account.clone())
+    }
+}
+
+impl<'a, L: Ledger, Key: KeyPair + DeserializeOwned + Serialize> Deref
+    for AccountEditor<'_, L, Key>
+{
+    type Target = Account<L, Key>;
+
+    fn deref(&self) -> &Account<L, Key> {
+        &self.account
+    }
+}
+
+impl<'a, L: Ledger, Key: KeyPair + DeserializeOwned + Serialize> DerefMut
+    for AccountEditor<'_, L, Key>
+{
+    fn deref_mut(&mut self) -> &mut Account<L, Key> {
+        &mut self.account
+    }
+}
+
+/// Accounts stored in an accounts store.
+pub struct Accounts<L: Ledger, Key: KeyPair + DeserializeOwned + Serialize> {
+    /// A key-value store for accounts.
+    store: AccountsStore<L, Key, Key::PubKey>,
+}
+
+impl<L: Ledger, Key: KeyPair + DeserializeOwned + Serialize> Accounts<L, Key> {
+    #![allow(dead_code)]
+
+    /// Load an accounts store.
+    pub fn new(store: AccountsStore<L, Key, Key::PubKey>) -> Result<Self, KeystoreError<L>> {
+        Ok(Self { store })
+    }
+
+    /// Iterate through the accounts.
+    pub fn iter(&self) -> impl Iterator<Item = Account<L, Key>> + '_ {
+        self.store.iter().cloned()
+    }
+
+    /// Get the account by the public key from the store.
+    pub fn get(&self, pub_key: &Key::PubKey) -> Result<Account<L, Key>, KeystoreError<L>> {
+        Ok(self.store.load(pub_key)?)
+    }
+
+    /// Get a mutable account editor by the public key from the store.
+    pub fn get_mut(
+        &mut self,
+        pub_key: &Key::PubKey,
+    ) -> Result<AccountEditor<L, Key>, KeystoreError<L>> {
+        let account = self.get(pub_key)?;
+        Ok(AccountEditor::new(&mut self.store, account))
+    }
+
+    /// Commit the store version.
+    pub fn commit(&mut self) -> Result<(), KeystoreError<L>> {
+        Ok(self.store.commit_version()?)
+    }
+
+    /// Revert the store version.
+    pub fn revert(&mut self) -> Result<(), KeystoreError<L>> {
+        Ok(self.store.revert_version()?)
+    }
+
+    /// Create an account.
+    ///
+    /// Returns the editor for the created account.
+    pub fn create(
+        &mut self,
+        key: Key,
+        description: String,
```

Comment:
Could probably make `description` optional and default it to `""` or `key.pub_key().to_string()` or something.